lib/Migration/NewDbStructureRepairStep.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace OCA\Tables\Migration;
+
+use OCA\Tables\Db\LegacyRowMapper;
+use OCA\Tables\Db\Row;
+use OCA\Tables\Db\RowMapper;
+use OCA\Tables\Db\Table;
+use OCA\Tables\Errors\InternalError;
+use OCA\Tables\Errors\PermissionError;
+use OCA\Tables\Service\ColumnService;
+use OCA\Tables\Service\TableService;
+use OCP\DB\Exception;
+use OCP\Migration\IOutput;
+use OCP\Migration\IRepairStep;
+use Psr\Log\LoggerInterface;
+
+class NewDbStructureRepairStep implements IRepairStep {
+
+	protected LoggerInterface $logger;
+	protected TableService $tableService;
+	protected LegacyRowMapper $legacyRowMapper;
+	protected RowMapper $rowMapper;
+	protected ColumnService $columnService;
+
+	public function __construct(LoggerInterface $logger, TableService $tableService, ColumnService $columnService, LegacyRowMapper $legacyRowMapper, RowMapper $rowMapper) {
+		$this->logger = $logger;
+		$this->tableService = $tableService;
+		$this->columnService = $columnService;
+		$this->legacyRowMapper = $legacyRowMapper;
+		$this->rowMapper = $rowMapper;
+	}
+
+	/**
+	 * Returns the step's name
+	 */
+	public function getName() {
+		return 'Copy the data into the new db structure';
+	}
+
+	/**
+	 * @param IOutput $output
+	 */
+	public function run(IOutput $output) {
+		$output->info("Look for tables");
+		try {
+			$tables = $this->tableService->findAll('', true, true, false);
```

Comment:
From a quick look I'd expect the userId needing to be null so the mapper selects all user tables?